sdk/core/core/src/az_http_response.c

```diff
@@ -258,3 +259,12 @@ AZ_NODISCARD az_result az_http_response_get_body(az_http_response* response, az_
   response->_internal.parser.next_kind = _az_HTTP_RESPONSE_KIND_EOF;
   return AZ_OK;
 }
+
+void _az_http_response_reset(az_http_response* http_response)
+{
+  // never fails, discard the result
+  // init will set written to 0 and will use the same az_span. Internal parser's state is also
+  // reset
+  az_result result = az_http_response_init(http_response, http_response->_internal.http_response);
```

Comment:
Can we add a precondition after this line that validates result is AZ_OK? This way, we catch unintentional bugs in the code and capture the intent in code.

Reply:
it never fails because it uses the same az_span that the http_response already have, so there are two options,
1) az_http_response_init() was not called before, in that case, http_response has an empty az_span internally anyway that gets created upon creation, so this function would have no effect on it
2) az_http_response_init() was called before. Then, the only thing that might change is the written variable if somethings was written before. Otherwise, is the same as previous casa

Reply:
I understand why it can never fail (with the current implementation). My point is, if the implementation of `az_http_response_init` were to change in the future, where it could fail, how would we make sure that this call site continues to work without an unintentional bug? The precondition would make sure of that and help catch the issue.

Reply:
If we have to, we'll repeat the portion of code that is in init that does a reset into this function.